pages/[id]/bridge.tsx

```diff
@@ -299,12 +306,25 @@ const Bridge = () => {
               </div>
               <div className="text-center">
                 {mounted &&
-                /* ⚪️ First, if we are on the wrong network, prompt to switch networks. */
+                /* ⚪️ First, if we are on the wrong network, prompt to switch networks. Add new network to user wallet if needed. */
                 source.chain.id !== chain?.id ? (
                   <button
                     type="button"
                     className="mt-5 rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 disabled:bg-gray-300 disabled:cursor-not-allowed"
-                    onClick={() => walletClient?.switchChain({ id: source.chain.id })}
+                    onClick={async () => {
```

Comment:
a few Qs here
1. I like that we're adding the chain! Is there an easy way to refactor though so we can use this on the bridge page too?
1. does `InternalRpcError` really imply that the wallet does not have target chain? seems too general.
1. Shouldn't we `switchChain` after we `addChain`? Or is that automatically happening?
1. Is the error handling on 323 appropriate? We are currently suppressing the error. Under what circumstances will we throw this error and do we care?

Reply:
1. Yes, I made it a custom hook in [d459419](https://github.com/ScopeLift/l2-flexible-voting-frontend/pull/133/commits/d4594196ee44dbf420cbe2d9a0d141a5655889b3)
2. You're right, seems like the error type that is relevant is just the `SwitchChainError`. I've removed `InternalRpcError` in the same commit as above.
3. I'm not sure why, maybe it's the wallet, but it's automatically switching to the new chain after being added.
4. Only scenario that I could think of was if the user rejects the request to add the new network to their wallet.

Reply:
Actually added the function to @util [2820cbc](https://github.com/ScopeLift/l2-flexible-voting-frontend/pull/133/commits/2820cbca985755420de837fe2d3b2c98656cc2ec) because React Hooks must be called in a React function component or a custom React Hook function error. Lmk if I did it right.